mteb/evaluation/evaluators/RetrievalEvaluator.py

```diff
@@ -277,6 +286,13 @@ def predict(self, queries, passages, **kwargs):
             "You must implement a predict method for your reranker model"
         )
 
+    def encode_conversations(self, conversations: List[List[str]], **kwargs):
+        if is_conv_retrieval_compatible(self.model):
```

Comment:
Based on my understanding, we will reach here if the user model has `encode` and `encode_conversations` implementation. The user model will get wrapped in `DRESModel` wrapper. The way `encode` function of DRESModel points to `encode` of user model, similarly, I think the `encode_conversations` of DRESModel should point to user model's implementation as it gives the user maximum flexibility to implement the logic to generate an embedding from conversation history.

Reply:
Sorry, I'm having a hard time explaining this.  We need to have a default function for `encode_conversations` that is implemented in the base classes `DRESModel` etc.  This way if the user starts it from the command line, it will work. 

But if the user extends the class and provides a new version of function, it should override that function.  So we don't have to check to be sure it's compatible since it will have extended the base class and will skip this function.

This also might become more clear if you add a task and then try extending `encode_conversations`.

Reply:
Let me know if my understanding here is correct

When the user model implements `encode_conversations`, it means that with the context of RetrievalEvaluator, the user model is either `self.retriever.model` (DRES compatible or cross encoder) or `self.retriever.model.model`.

If this is correct, then the current implementation will use user's implementation whether user model extends DRESModel or not.

As you suggested, I'll verify this by extending `encode_conversations`.

Reply:
@KennethEnevoldsen We had a chat about the above and I realized I may have had a misconception about the way people add custom functions.

From the README, someone can add the `encoder_queries` function without extending `DRESModel` or `DenseRetrievalExactSearch` correct?  In that case we will leave the code as @vaibhavad set it.

Reply:
@orionw yes that is the intention at least